src/TTImp/Elab/App.idr

```diff
@@ -330,6 +335,20 @@ mutual
                       else pure $ dotTerm tm
            else pure tm
     where
+      
+      ||| Find the constructors of a normal form that is a fully
+      ||| applied type constructor
+      possibleConstructors : NF vars -> Core (List Name)
+      possibleConstructors (NApp _ (NRef (TyCon _ n) tycName) args) = 
+        if length args == n
```

Comment:
This is one reason the tests are failing - `NApp` is for stuck function applications, whereas type constructors are already in normal form, so not stuck. Instead, you need `NTCon`

Reply:
Thanks, sorry for misunderstanding, and it makes sense.